includes/class-amp-travel-cpt.php

```diff
@@ -0,0 +1,288 @@
+			'uploaded_to_this_item' => _x( 'Uploaded to this adventure', 'Overrides the “Uploaded to this post”/”Uploaded to this page” phrase (used when viewing media attached to a post).', 'travel' ),
+			'filter_items_list'     => _x( 'Filter adventures list', 'Screen reader text for the filter links heading on the post type listing screen. Default “Filter posts list”/”Filter pages list”.', 'travel' ),
+			'items_list_navigation' => _x( 'Adventures list navigation', 'Screen reader text for the pagination heading on the post type listing screen. Default “Posts list navigation”/”Pages list navigation”.', 'travel' ),
+			'items_list'            => _x( 'Adventures list', 'Screen reader text for the items list heading on the post type listing screen. Default “Posts list”/”Pages list”.', 'travel' ),
+		);
+
+		$args = array(
+			'labels'                => $labels,
+			'description'           => __( 'Adventure Custom Post Type for travel theme.', 'travel' ),
+			'public'                => true,
+			'exclude_from_search'   => true,
+			'publicly_queryable'    => true,
+			'show_ui'               => true,
+			'show_in_nav_menus'     => true,
+			'show_in_menu'          => true,
+			'show_in_admin_bar'     => true,
+			'menu_position'         => 20,
+			'menu_icon'             => 'dashicons-location-alt',
+			'capability_type'       => 'post',
+			'hierarchical'          => false,
+			'supports'              => array(
+				'title',
+				'editor',
+				'thumbnail',
+			),
+			'has_archive'           => true,
+			'rewrite'               => array(
+				'slug' => self::POST_TYPE_SLUG_SINGLE,
+			),
+			'query_var'             => true,
+			'can_export'            => true,
+			'show_in_rest'          => true,
+			'rest_base'             => self::POST_TYPE_SLUG_PLURAL,
+			'rest_controller_class' => 'WP_REST_Posts_Controller',
+		);
+
+		register_post_type( self::POST_TYPE_SLUG_SINGLE, $args );
+	}
+
+	/**
+	 * Adds meta boxes for adventure post type.
+	 */
+	public function add_adventure_meta_boxes() {
+		add_meta_box( 'amp_travel_adventure_meta', __( 'Adventure details' ), array( $this, 'adventure_meta_box_html' ), 'adventure', 'side' );
+	}
+
+	/**
+	 * Displays meta boxes in admin.
+	 */
+	public function adventure_meta_box_html() {
+		global $post;
+		$adventure_custom = get_post_custom( $post->ID );
+		$price            = isset( $adventure_custom['amp_travel_price'][0] ) ? $adventure_custom['amp_travel_price'][0] : '';
+		?>
+		<label for='amp_travel_price'><?php esc_attr_e( 'Price (USD)', 'travel' ); ?></label>
+		<input id='amp_travel_price' name='amp_travel_price' value='<?php echo esc_attr( $price ); ?>'>
+		<?php wp_nonce_field( basename( __FILE__ ), 'amp_travel_price_nonce' ); ?>
+		<?php
+	}
+
+	/**
+	 * Saves the custom meta.
+	 */
+	public function save_adventure_post() {
+		if ( ! empty( $_POST ) ) {
```

Comment:
This might not be needed. If `$_POST` is empty, the function will return in the next conditional when `isset( $_POST['amp_travel_price'] )` is `false`.